### PR TITLE
Implemented redux store for ordering component

### LINKED ISSUE #93: Create redux store for order component

### PR DESCRIPTION
**Pull request by nasaku898**
resolves #93 

Features remain the same, but the data is now store in redux state instead of react state. This makes it easier to access data and manage state across the application.

### CODE REVIEW

**Review by Mnx458 — APPROVED** [review]

File: bike-erp/backend/src/main/routes/component_routes.ts [hunk 1]
```diff
@@ -39,6 +39,16 @@ router.put("/updateQuantity", (req, res) => {
     });
 });
 
+router.put("/orderComponents", (req, res) => {
```
Comment (Manraj45) [inline bike-erp/backend/src/main/routes/component_routes.ts:42]: add comments

File: bike-erp/backend/src/main/services/inventoryManagementService/InventoryManagementService.ts [hunk 2]
```diff
@@ -32,6 +32,18 @@ export class InventoryManagementService {
     return updateComponent(id, quantity);
   };
 
+  public orderComponents = (orderList: Array<any>) => {
```
Comment (Manraj45) [inline bike-erp/backend/src/main/services/inventoryManagementService/InventoryManagementService.ts:35]: add comments

File: bike-erp/src/redux/actions/OrderListActions/orderListAction.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import {ADD_ITEM, REMOVE_ALL_ITEMS, REMOVE_ITEM} from '../../types/OrderListTypes/orderListTypes'
+
```
Comment (Manraj45) [inline bike-erp/src/redux/actions/OrderListActions/orderListAction.ts:2]: add comments

File: bike-erp/src/redux/reducers/OrderListReducers/orderListReducer.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { ADD_ITEM, REMOVE_ALL_ITEMS, REMOVE_ITEM } from "../../types/OrderListTypes/orderListTypes";
+import { Order } from "../../actions/OrderListActions/orderListAction"
+const initialState = {
+    error: "",
+    orderList: []
+}
+
+const reducer = (state = initialState, action: any) => {
```
Comment (Manraj45) [inline bike-erp/src/redux/reducers/OrderListReducers/orderListReducer.ts:8]: add comments

**Review by Manraj45 — CHANGES_REQUESTED** [review]
When the user is inactive, he is redirected to the login page. However, the user can still access the order and inventory page from the login page. This bug must be fixed.

File: bike-erp/src/redux/actions/OrderListActions/orderListAction.ts (hunk 3 above)
Comment (nasaku898) [reply bike-erp/src/redux/actions/OrderListActions/orderListAction.ts:2]: The redux action are self explanatory based on function names

**Review by Manraj45 — APPROVED** [review]
Good job! Created a different issue for the problem

**Pull request merged**